kinto/plugins/history/views.py

```diff
@@ -43,3 +45,86 @@ def _extract_filters(self, queryparams=None):
             filters_str_id.append(filt)
 
         return filters_str_id
+
+
+class Revert(object):
+
+    CREATE = 'create'
+    UPDATE = 'update'
+    DELETE = 'delete'
+
+    def __init__(self, request):
+        self.request = request
+
+    def revert_changes(self, history, since):
+        revert_list = self.build_revert_list(history, since)
+        if len(revert_list) == 0:
+            raise_error(self.request, 'no history', 'there are no history records to revert after since timestamp')
+        return { 'data': revert_list }
+
+    def create_revert_operation(self, last, first, passed_since):
+        rec_id = last['target']['data']['id']
+        if last['action'] == self.CREATE or (not passed_since and first['action'] == self.CREATE):
+            return {'action': self.DELETE, 'data':  {'id': rec_id}}
+        else:
+            return {'action': self.UPDATE, 'data':  first['target']['data']}
+
+    def insert_rec_to_revert_lists(self, rec, revert_lists, passed_since):
+        rec_id = rec['target']['data']['id']
+        if passed_since:
+            if rec_id not in revert_lists['not_paired']:
+                return
+            last_change = revert_lists['not_paired'].pop(rec_id)
+            revert_lists['paired'].append(self.create_revert_operation(last_change, rec, passed_since))
+        else:
+            if rec_id in revert_lists['not_paired']:
+                if rec['action'] == self.CREATE:
+                    last_change = revert_lists['not_paired'].pop(rec_id)
+                    revert_lists['paired'].append(self.create_revert_operation(last_change, rec, passed_since))
+            elif rec['action'] == self.CREATE:
+                revert_lists['paired'].append(self.create_revert_operation(rec, None, passed_since))
+            else:
+                revert_lists['not_paired'][rec_id] = rec
+
+    def build_revert_list(self, history, since):
+        revert_lists = {'not_paired': {}, 'paired': []}
+        passed_since = False
+        for rec in history['data']:
+            if not passed_since and rec['last_modified'] < since:
+                passed_since = True
+            print passed_since
+            if passed_since and len(revert_lists['not_paired']) == 0:
+                break
+            self.insert_rec_to_revert_lists(rec, revert_lists, passed_since)
+
+        return revert_lists['paired']
+
+
+revert_service = Service(name="revert",
+                     path='/buckets/{bucket_id}/collections/{collection_id}/revert',
```

Comment:
I think the revert operation should be at the same level as history. Thus `/buckets/{bucket_id}/revert`.